lib/takeaway.rb

```diff
@@ -0,0 +1,49 @@
+require_relative 'menu'
+require_relative 'order'
+
+class Takeaway
+  attr_reader :menu, :order, :total
+
+  ERROR = "Total cost does not match the sum of the dishes in your order!"
+
+  def initialize(menu_klass = Menu.new, order_klass = Order.new)
+    @menu = menu_klass
+    @order = order_klass
+  end
+
+  def open_menu
+    menu.open
+  end
+
+  def order(item, quantity = 1)
+    order.add_basket(item, quantity)
+    "#{quantity}x #{item}(s) added to your basket."
+  end
+
+  def order_summary
+    order.basket_summary(menu)
+  end
+
+  def total
+    @total = order.total(menu) unless empty?
+    "Total: £#{sprintf('%.2f', @total)}"
+  end
+
+  def checkout(amount = 0)
```

Comment:
Lint/UnusedMethodArgument: Unused method argument - `amount`. If it's necessary, use `_` or `_amount` as an argument name to indicate that it won't be used. You can also write as `checkout(*)` if you want the method to accept any arguments but don't care about them. (https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)